library/time-library/src/main/java/uk/gov/gchq/gaffer/time/function/DateToTimeBucketEnd.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.time.function;
+
+import uk.gov.gchq.gaffer.time.CommonTimeUtil;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.function.KorypheFunction;
+
+import java.util.Date;
+
+import static java.util.Objects.isNull;
+
+/**
+ * Converts a Date into the end of a timestamp bucket, based on a provided
+ * {@link CommonTimeUtil.TimeBucket}.
+ */
+@Since("1.21.0")
+@Summary("Converts a Date into the end of a timestamp bucket, based on a provided TimeBucket")
+public class DateToTimeBucketEnd extends KorypheFunction<Date, Date> {
+    private CommonTimeUtil.TimeBucket bucket;
+
+    public DateToTimeBucketEnd() {
+    }
+
+    public DateToTimeBucketEnd(final CommonTimeUtil.TimeBucket bucket) {
```

Comment:
It might be a bit easier to read if these references were made less explicit (via import)

Reply:
I agree it looks ugly but there are other places already in the codebase where it is done like this: https://github.com/gchq/Gaffer/blob/fc9fe8e52ab64667ca5fbec80deef0a5ac36a27a/library/time-library/src/main/java/uk/gov/gchq/gaffer/time/serialisation/BoundedTimestampSetSerialiser.java#L82

Should I change it everywhere or just the files being added in this PR?